internal/admin/handlers.go

```diff
@@ -0,0 +1,573 @@
+			Version:      version.Get().EnvoyGatewayVersion,
+			Uptime:       uptime,
+			Status:       "Running",
+			ConfigStatus: "Loaded",
+			Config:       cfg.ServerConfiguration,
+		}
+
+		if err := json.NewEncoder(w).Encode(response); err != nil {
+			http.Error(w, fmt.Sprintf("Failed to encode response: %v", err), http.StatusInternalServerError)
+		}
+	}
+}
+
+// handleResources returns information about Gateway resources
+func handleResources(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	if k8sClient == nil {
+		http.Error(w, "Kubernetes client not available", http.StatusServiceUnavailable)
+		return
+	}
+
+	ctx := context.Background()
+	response := make(ResourcesResponse)
+
+	// Get GatewayClasses
+	if gatewayClasses, err := getGatewayClasses(ctx); err == nil {
+		response["GatewayClasses"] = gatewayClasses
+	}
+
+	// Get Gateways
+	if gateways, err := getGateways(ctx); err == nil {
+		response["Gateways"] = gateways
+	}
+
+	// Get HTTPRoutes
+	if httpRoutes, err := getHTTPRoutes(ctx); err == nil {
+		response["HTTPRoutes"] = httpRoutes
+	}
+
+	// Get GRPCRoutes
+	if grpcRoutes, err := getGRPCRoutes(ctx); err == nil {
+		response["GRPCRoutes"] = grpcRoutes
+	}
+
+	// Get TCPRoutes
+	if tcpRoutes, err := getTCPRoutes(ctx); err == nil {
+		response["TCPRoutes"] = tcpRoutes
+	}
+
+	// Get UDPRoutes
+	if udpRoutes, err := getUDPRoutes(ctx); err == nil {
+		response["UDPRoutes"] = udpRoutes
+	}
+
+	// Get TLSRoutes
+	if tlsRoutes, err := getTLSRoutes(ctx); err == nil {
+		response["TLSRoutes"] = tlsRoutes
+	}
+
+	// Get BackendTLSPolicies
+	if backendTLSPolicies, err := getBackendTLSPolicies(ctx); err == nil {
+		response["BackendTLSPolicies"] = backendTLSPolicies
+	}
+
+	// Get BackendTrafficPolicies
+	if backendTrafficPolicies, err := getBackendTrafficPolicies(ctx); err == nil {
+		response["BackendTrafficPolicies"] = backendTrafficPolicies
+	}
+
+	// Get ClientTrafficPolicies
+	if clientTrafficPolicies, err := getClientTrafficPolicies(ctx); err == nil {
+		response["ClientTrafficPolicies"] = clientTrafficPolicies
+	}
+
+	// Get SecurityPolicies
+	if securityPolicies, err := getSecurityPolicies(ctx); err == nil {
+		response["SecurityPolicies"] = securityPolicies
+	}
+
+	// Get EnvoyPatchPolicies
+	if envoyPatchPolicies, err := getEnvoyPatchPolicies(ctx); err == nil {
+		response["EnvoyPatchPolicies"] = envoyPatchPolicies
+	}
+
+	// Get EnvoyExtensionPolicies
+	if envoyExtensionPolicies, err := getEnvoyExtensionPolicies(ctx); err == nil {
+		response["EnvoyExtensionPolicies"] = envoyExtensionPolicies
+	}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		http.Error(w, fmt.Sprintf("Failed to encode response: %v", err), http.StatusInternalServerError)
+	}
+}
+
+// handleStats returns statistics information
+func handleStats(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var memStats runtime.MemStats
```

Comment:
nit: can be a follow up, https://pkg.go.dev/runtime/metrics seems to expose more, which can be used to also export to prom